### DISCUSSION TITLE
Wildcard mappings / Wrap in RoutingServlet

### DISCUSSION THREAD

**Discussion by thiccaxe** [activej/activej#152] [Q&A] [answered]
Hi, I am a bit confused on how I should properly have "default" / 404 handlers in a RoutingServllet.

Consider the following example:
```java
RoutingServlet
        .wrap(request -> notFound404().withPlainText("ABC...1"))
        .map("/api", RoutingServlet
                .wrap(request -> notFound404().withPlainText("DEF...2"))
                .map("/route", request -> ok200().withPlainText("route"))
        )
;
```
`/` results in "ABC...1" as expected, and `/api` results in "DEF...2" as well. However, when i try and request `/api/route`, I see that I get "ABC...1". Am I misunderstanding the purpose of wrap? (From the code it looks like it sets the default/fallback handler)

I tried similarly using wildcards, essentially replacing as such:
```java
RoutingServlet.create()
        .map("/api", RoutingServlet.create()
                .map("/route", request -> ok200().withPlainText("route"))
                .map("/*", request -> notFound404().withPlainText("DEF...2")))
        .map("/*", request -> notFound404().withPlainText("ABC...1"))
;
```

Can someone provide clarification on how to use the RoutingServlet for a default route?
Thanks.

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @thiccaxe

The second approach is preferred for handling 404s. 

You actually need to replace the `"/api"` routing pattern with the `"/api/*"`. The added asterisk means that there are more URL path parts expected after `/api/`. You can add the asterisk to any of the provided snippets and `/api/route` request should resolve to the correct servlet.

**Comment by thiccaxe** [comment]
Hey, thanks!
@eduard-vasinskyi  I was wondering if you guys were open to the github discussions for these types of questions so real bugs/problems can stay in the issues tab.
